streaming/index.js

```diff
@@ -183,6 +184,73 @@ const startServer = async () => {
   const redisSubscribeClient = await redisUrlToClient(redisParams, redisUrl);
   const redisClient = await redisUrlToClient(redisParams, redisUrl);
 
+  // Collect metrics from Node.js
+  metrics.collectDefaultMetrics();
+
+  new metrics.Gauge({
+    name: 'pg_pool_total_connections',
+    help: 'The total number of clients existing within the pool',
+    collect() {
+      this.set(pgPool.totalCount);
+    },
+  });
+
+  new metrics.Gauge({
+    name: 'pg_pool_idle_connections',
+    help: 'The number of clients which are not checked out but are currently idle in the pool',
+    collect() {
+      this.set(pgPool.idleCount);
+    },
+  });
+
+  new metrics.Gauge({
+    name: 'pg_pool_waiting_queries',
+    help: 'The number of queued requests waiting on a client when all clients are checked out',
+    collect() {
+      this.set(pgPool.waitingCount);
+    },
+  });
+
+  const connectedClients = new metrics.Gauge({
+    name: 'connected_clients',
+    help: 'The number of clients connected to the streaming server',
+    labelNames: ['type'],
+  });
+
+  connectedClients.set({ type: 'websocket' }, 0);
+  connectedClients.set({ type: 'eventsource' }, 0);
+
+  const connectedChannels = new metrics.Gauge({
+    name: 'connected_channels',
+    help: 'The number of channels the streaming server is streaming to',
+    labelNames: [ 'type', 'channel' ]
```

Comment:
Note: `channel` here is from a bounded set, so the maximum number of labels will limited to:
- user
- user:notification
- public
- public:local
- public:remote
- public:media
- public:local:media
- public:remote:media
- hashtag
- hashtag:local
- direct
- list

This does not leak, for instance, User IDs into the metrics.

Reply:
I'm not familiar with prometheus or `prom-client`, but doesn't this break compatibility with the previous format?

Reply:
I think it should be fine, as it's still a Guage, it just has additional data associated with it. From the prometheus documentation:
> If you are unsure, start with no labels and add more labels over time as concrete use cases arise.

— https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels